### PR TITLE
Dm 2726 - Disable ability to comment for public users.

### PR DESCRIPTION
**Pull request by bradjohnson92008**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2726

## Description - what does this code do?
This code identifies if the current user is a "public" user and if so disables the ability to comment or reply to the Practice comment's section on the Practice show page.  User can still view previous comments but comments are read only and they can not create new comments on the practice.  Edit and delete links are disabled.

## Testing done - how did you test it/steps on how can another person can test it 
1.  Login to DM as a a non "ntlm" user.  i.e., non-gfe or cag user.  Following method should return nil for user in the application controller.  This will set the current user to a "public" user.
image
2.  Navigate to a Practice show page that contains comments.  
3. Scroll down to the pages "Contact / Comment" section.  Since you are logged in as a public user you should see the INformation banner stating that "Comments and replies are disabled for retired practices and public users."
image
4.  Scroll through the comments.  User should be able to view all comments, and expand or collapse "Replies".
5. Click on the Edit and Delete icons.  These icons should be disabled - i.e., no action occurs when clicking them.
6.  Log off and log back on as an NTLM (VHA network) user.  
7. Navigate to a practice that is not retired.
8. Comment and reply functionality should enabled and all functionality wihin commontator should work correctly.

## Acceptance criteria
Public users don't see comment textbox or role radio button
Comments are read only
Test on Dev when public
Test on STG when public
Release to Prod

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by emyl3 — CHANGES_REQUESTED** [review]
The styles are looking a little off for me. Is anyone else seeing this too?
Screen Shot 2021-07-20 at 4 01 48 PM


Also, I am seeing the edit button when I'm a different user for a comment that I didn't write and this user is not an admin.
Screen Shot 2021-07-20 at 4 20 53 PM
Should we just hide the "Reply" button and thumbs up button if they are a public user?

File: app/views/commontator/comments/_form.html.erb [hunk 1]
```diff
@@ -33,70 +33,72 @@
...
-        </div>
-        <div class="usa-radio">
-          <%= radio_button_tag 'user_practice_status', :team_member, practice_team_member ? true : false, class: 'usa-radio__input', id: "team-member-#{comment.id}" %>
-          <%= label_tag 'user_practice_status_team_member', 'I am a member of this practice team', class: 'usa-radio__label team-member-label', for: "team-member-#{comment.id}" %>
-        </div>
-      <% end %>
-    </div>
-    <div class="text-left margin-top-2">
-      <% if config.new_comment_style == :t && new_record && comment.parent.nil? %>
+    <% is_public_user = session[:user_type] === 'public' ? true : false %>
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply app/views/commontator/comments/_form.html.erb:36]: true.. :)

File: db/schema.rb [hunk 2]
```diff
@@ -10,8 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2021_06_02_143652) do
-
+ActiveRecord::Schema.define(version: 2021_06_10_160757) do
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply db/schema.rb]: Got it.

**Comment by JoshingYou1** [comment]
> The styles are looking a little off for me. Is anyone else seeing this too?
Yeah, the styling is off on my local as well
Screen Shot 2021-07-21 at 2 00 14 PM

> Should we just hide the "Reply" button and thumbs up button if they are a public user?
I think that's a good idea. We should hide anything the user cannot actually interact with

File: db/schema.rb [hunk 3]
```diff
@@ -217,6 +216,10 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "clinical_id", id: false, force: :cascade do |t|
+    t.bigint "id"
+  end
+
```
Comment (JoshingYou1) [inline db/schema.rb:222]: Are we using this?

**Review by JoshingYou1 — CHANGES_REQUESTED** [review]
Aside from some of the styling, just left one comment Brad. Nice work!

File: db/schema.rb (hunk 3 above)
Comment (bradjohnson92008) [reply db/schema.rb]: No - I removed that create_table code

**Comment by bradjohnson92008** [comment]
Added more code to remove all the icons/buttons ..etc that the user can not interact with.  Also removed link for comment creator - you can still see the comment creators name and job title.. but removed the link to the profile page.

**Review by emyl3 — CHANGES_REQUESTED** [review]
Found a couple of lines of code we can remove.

Thoughts on left aligning the "Show X reply" button. It looks off not being left aligned without the other buttons.
Screen Shot 2021-07-22 at 10 17 18 AM

File: app/views/commontator/comments/_show.html.erb [hunk 4]
```diff
@@ -93,33 +95,34 @@
                              class: is_deleted ? "fas fa-trash-restore-alt fa-lg margin-left-1" : "fas fa-trash-alt fa-lg delete-comment-icon margin-left-1",
                              remote: true %>
             <% else %>
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply app/views/commontator/comments/_show.html.erb:97]: Got it.

File: app/views/commontator/comments/_show.html.erb [hunk 5]
```diff
@@ -147,9 +150,9 @@
               %>
             </span>
             <% else %>
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply app/views/commontator/comments/_show.html.erb:152]: Got it.

**Comment by bradjohnson92008** [comment]
I left aligned the Show/Hide Replies btn/link since we are not displaying the other icons.

**Review by emyl3 — APPROVED** [review]
Looks good to me! Great work, Brad!

**Review by JoshingYou1 — APPROVED** [review]
Works as intended. Great job Brad!!

**Pull request merged**